### PR TITLE
Add SQL fixes for overtime incident Jan 13 - Jan 18 2022

### PR DESCRIPTION
**Pull request by NeilvB**
## Link to JIRA ticket

This forms the Repairs API portion of https://www.pivotaltracker.com/n/projects/2500129/stories/180964019

### *What is the problem we're trying to solve*

Following overtime incident Jan 13 2022 - Jan 18 2022 where all work orders submitted by operatives were set as overtime due to a bug in the frontend.

The PR comprises two commits:

1. In-hours jobs where operatives should not be able to set overtime at these times - we know that this set needs to be updated to non-overtime. [Commit](https://github.com/LBHackney-IT/repairs-api-dotnet/pull/566/commits/a1033f49d6144ee5bd1d66905df352f7d58eb2d1).
2. Outside normal hours jobs where operatives could have intended to submit an order as non-overtime. For this set we are reliant on the operatives having told us which historical jobs were overtime or not. [Commit](https://github.com/LBHackney-IT/repairs-api-dotnet/pull/566/commits/0d09a21043ac8f49f04b63c2b4ea617eda1054fe).

The queries in the second commit assumes the presence of a temporary table with information about the intended payment type for each period, taking the following format:

```
+--------------+--------------------------------+--------------------------------+-------------------------------+-------------------------------+--------------------------------+-------------------------------+-------------------------------+--------------------------------+--------------------------------+
| operative_id | thusday_13_jan_am_payment_type | thusday_13_jan_pm_payment_type | friday_14_jan_am_payment_type | friday_14_jan_pm_payment_type | weekend_15_16_jan_payment_type | monday_17_jan_am_payment_type | monday_17_jan_pm_payment_type | tuesday_18_jan_am_payment_type | tuesday_18_jan_pm_payment_type |
+--------------+--------------------------------+--------------------------------+-------------------------------+-------------------------------+--------------------------------+-------------------------------+-------------------------------+--------------------------------+--------------------------------+
|          101 | bonus                          | bonus                          | bonus                         | bonus                         | bonus                          | bonus                         | bonus                         | bonus                          | bonus                          |
|          102 | bonus                          | bonus                          | bonus                         | bonus                         | bonus                          | bonus                         | bonus                         | bonus                          | bonus                          |
+--------------+--------------------------------+--------------------------------+-------------------------------+-------------------------------+--------------------------------+-------------------------------+-------------------------------+--------------------------------+--------------------------------+
```

### *What changes have we introduced*

Data recovery

#### _Checklist_

- [ ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [ ] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [ ] Checked all code for possible refactoring
- [ ] Code pipeline builds correctly

### *Follow up actions after merging PR*

 - [ ] Ensure we have records of all affected work orders for follow up recovery actions in Bonus Calculation
 - [ ] Run the update versions of the queries on production

### CODE REVIEW

File: sql/overtime-incident-jan-13-2022-data-recovery/correct-in-hours-job-status-update-comments.sql [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+FROM
+	job_status_updates
+--UPDATE
+--	job_status_updates
+--SET
+--	-- substring up until the start of the suffix ' - Overtime' which is 11 characters
+--	comments = substring(comments, 0, length(comments) - 10)
+WHERE
+	-- overtime problem period, further scoped below
+	event_time >= '2022-01-13 18:25:47' AND event_time <= '2022-01-18 18:06:01'
```
Comment (Asura5) [inline sql/overtime-incident-jan-13-2022-data-recovery/correct-in-hours-job-status-update-comments.sql:13]: Why is this necessary if there are more specific datetime filters below?

Comment (NeilvB) [reply sql/overtime-incident-jan-13-2022-data-recovery/correct-in-hours-job-status-update-comments.sql:13]: It isn't - I just kept this in because I felt it added to the readability to have an overall upper and lower limit which is then filtered further in later parts.

**Review by Asura5 — APPROVED** [review]
Looks good pending testing on the prod copy

**Comment by pixeltrix** [comment]
This PR isn't affected but it's worth remembering that this is a public repo and we should make sure we don't use PII data in any of these SQL files. Even if it's spotted in the PR the commit is still reachable and we'd need to raise a ticket with GitHub support - treat it like you would a access token.

**Review by pixeltrix — APPROVED** [review]

**Review by teodor-hristoforov-hackney — APPROVED** [review]

**Review by pixeltrix — APPROVED** [review]

**Comment by NeilvB** [comment]
Run on the production copy.

In-hours work orders and associated job status updates, all currently overtime, all of which are to be set to bonus
UPDATE 371

Out of the possibly-overtime hours work orders, all currently overtime which are to be set to bonus based on information from the business
UPDATE 69
